components/textdiff/textdiff.js

```diff
@@ -79,16 +75,15 @@ class TextDiffViewModel {
     this.htmlSrc = undefined;
     this.isParsed = ko.observable(false);
 
-    programEvents.add(event => {
-      if (event.event === "invalidate-diff-and-render" || event.event === "working-tree-changed") {
-        this.invalidateDiff();
-        if (this.isShowingDiffs()) this.render();
-      }
-    });
-
     this.isShowingDiffs.subscribe(newValue => {
       if (newValue) this.render();
     });
+    this.textDiffType.value.subscribe(() => {
+      if (this.isShowingDiffs()) this.render();
```

Comment:
Does a change in type no longer invalidate the diffs?

Reply:
No, because it isn't needed.
The diff type setting and "load more" are client side features and don't require server requests.
The whitespace setting on the other hand is a git feature and requires a server request.